Source/EasyNetQ/Producer/DefaultPubSub.cs

```diff
@@ -121,5 +124,12 @@ CancellationToken cancellationToken
 
             return new SubscriptionResult(exchange, queue, consumerCancellation);
         }
+
+        private CancellationTokenSource CreateCancellationTokenSource(CancellationToken cancellationToken)
+        {
+            var cts = CancellationTokenSource.CreateLinkedTokenSource(cancellationToken);
+            cts.CancelAfter(connectionConfiguration.Timeout);
+            return cts;
+        }
```

Comment:
It eliminates internal Timer creation inside `CancelAfter`.

```suggestion
        private CancellationTokenSource CreateCancellationTokenSource(CancellationToken cancellationToken)
        {
            var cts = CancellationTokenSource.CreateLinkedTokenSource(cancellationToken);
            if (connectionConfiguration.Timeout != Timeout.InfitineTimeSpan)
                cts.CancelAfter(connectionConfiguration.Timeout);
            return cts;
        }
```